magma/interface.py

```diff
@@ -491,6 +491,10 @@ def __getattr__(self, key):
             return self._ports[key]
         return super().__getattribute__(key)
 
+    def flip(self):
```

Comment:
I think we should also make `flip(self) -> 'IOInterface'` a default method on `IOInterface` base class that raises `NotImplenetedError`. This way the derived classes don't implement them..